x/autopilot/types/parser.go

```diff
@@ -74,18 +56,25 @@ func (m ClaimPacketMetadata) Validate() error {
 
 // Parse packet metadata intended for autopilot
 // In the ICS-20 packet, the metadata can optionally indicate a module to route to (e.g. stakeibc)
-// The PacketForwardMetadata returned from this function contains attributes for each autopilot supported module
+// The AutopilotMetadata returned from this function contains attributes for each autopilot supported module
 // It can only be forward to one module per packet
-// Returns nil if there was no metadata found
-func ParsePacketMetadata(metadata string) (*PacketForwardMetadata, error) {
+// Returns nil if there was no autopilot metadata found
+func ParseAutopilotMetadata(metadata string) (*AutopilotMetadata, error) {
 	// If we can't unmarshal the metadata into a PacketMetadata struct,
 	// assume packet forwarding was no used and pass back nil so that autopilot is ignored
 	var raw RawPacketMetadata
 	if err := json.Unmarshal([]byte(metadata), &raw); err != nil {
 		return nil, nil
 	}
 
-	// If no forwarding logic was used for autopilot, return the metadata with each disabled
+	// Packets cannot be used for both autopilot and PFM at the same time
+	// If both fields were provided, reject the packet
+	if raw.Autopilot != nil && raw.Forward != nil {
```

Comment:
> Unwind a packet to Stride then use autopilot

I believe in this case, the memo would start as:
```json
{
  "forward": {
    "channel": "umee-to-osmo-channel",

    "next": {
      "forward": {
        "channel": "osmo-to-stride-channel",

        "next": {
          "autopilot": "..."
        }
```

And by the time it got to stride, it would be just
```json
{
   "autopilot": "..."
}
```

---

> I think there's also a slightly different case where Stride first is a hop on the PFM path, then the token lands back on Stride 

I can't think of any examples either. Imo, I think it's fine to reject this case. If you're attempting to use autopilot, I think we can assume the PFM'ing should only tough stride in one place, at the last hop.

Reply:
Sounds good and makes sense to me! Thanks for confirming